File: linked_list/reverse_linked_list.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *prev = nullptr;
        while (head != nullptr) {
            ListNode *x = head;
            head = head->next;
            x->next = prev;
            prev = x;
        }
        
        return prev;
    }
};

File: easy/linked_list/reverse_linked_list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        // https://www.geeksforgeeks.org/reverse-a-linked-list/
        ListNode curr = null;
        ListNode prev = null;
        ListNode next = null;
        curr = head;
        while (curr != null) {
            next = curr.next;
            curr.next = prev;
            prev = curr;
            curr = next;
        }
        return prev;
    }
}